File: ecommerce/src/main/java/com/huy/ecommerce/entities/Payment.java
package com.huy.ecommerce.entities;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;

@Entity
@Table(name = "payments")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long paymentId;

    @OneToOne(mappedBy = "payment")
    private Order order;

    private String paymentMethod;

    private String paymentStatus;
    private Double amount;

    @Temporal(TemporalType.TIMESTAMP)
    private Date paymentTime;

}

File: e-commerce-angular/src/app/shared/dto/payment.ts
export interface Payment {
    paymentId?: string;        // PayPal payment ID
    paymentMethod?: string;              // PayPal payer ID
    paymentStatus: string;    // Payment status (e.g., "Completed")
    paymentTime?: string;      // Time when the payment was made
    amount?: number;          // Payment amount
}
